orix/quaternion/orientation.py

```diff
@@ -178,6 +192,31 @@ def set_symmetry(self, Gl, Gr, verbose=False):
         [[-0.7071  0.7071  0.      0.    ]
         [ 0.      1.      0.      0.    ]]
         """
+        misori = self.__class__(self.data)
+        misori.symmetry = (Gl, Gr)
+        return misori.compute_symmetry_reduced_orientations()
+
+    def compute_symmetry_reduced_orientations(self, verbose=False):
+        """Computes equivalent transformations which have the smallest
+        angle of rotation and assigns these in-place.
+
+        Returns
+        -------
+        Misorientation
+            A new misorientation object with the assigned symmetry.
+
+        Examples
+        --------
+        >>> from orix.quaternion.symmetry import C4, C2
+        >>> data = np.array([[0.5, 0.5, 0.5, 0.5], [0, 1, 0, 0]])
+        >>> m = Misorientation(data)
+        >>> m.symmetry = (C4, C2)
+        >>> m.compute_symmetry_reduced_orientations()
+        Misorientation (2,) 4, 2
+        [[-0.7071  0.7071  0.      0.    ]
+        [ 0.      1.      0.      0.    ]]
+        """
+        Gl, Gr = self._symmetry
```

Comment:
Following the docstring, I find `assign_smallest_angle()` a more informative name, what do you think @harripj?

What do you think @pc494? You've touched this part of the code the most of us.

I've also suggested small changes to the docstring example, in anticipation of us testing these as part of CI (using `pytest --doctest`).

```suggestion
    def assign_smallest_angle(self, verbose=False):
        """Computes equivalent transformations which have the smallest
        angle of rotation and assigns these in-place.

        Returns
        -------
        Misorientation
            A new misorientation instance with the smallest angle of rotation.

        Examples
        --------
        >>> import numpy as np
        >>> from orix.quaternion import Misorientation, symmetry
        >>> data = np.array([[0.5, 0.5, 0.5, 0.5], [0, 1, 0, 0]])
        >>> m = Misorientation(data)
        >>> m.symmetry = (symmetry.C4, symmetry.C2)
        >>> m.assign_smallest_angle()
        Misorientation (2,) 4, 2
        [[-0.7071  0.7071  0.      0.    ]
        [ 0.      1.      0.      0.    ]]
        """
        Gl, Gr = self._symmetry
```

Reply:
The name of this function is certainly up for debate!

@hakonanes I think that the `smallest_angle` part is very informative, but to me `assign` indicates assign in-place rather than return a new instance (which is what the function is currently doing). I would prefer `compute` or `calculate` to this end. What do you think?

Reply:
I agree that by assign we mean in-place, which is what the docstring says, but actually doesn't do! Another place where this docstring could be improved, good catch indeed.

I think @pc494's suggestion of `map_into_symmetry_reduced_zone()` (or perhaps `map_into_fundamental_zone()`) is even better, what do you think?

Reply:
Yes, well spotted! I will update the docstring here.

I think that both of these suggestions are informative and make sense. `map_into_symmetry_reduced_zone()` is easy to understand. `map_into_fundamental_zone()` I think we have to be clear that this is what is actually happening, and what would distinguish `zone` from `region` (as in `OrientationRegion`)?

Reply:
Let's use map_to_symmetry_reduced_zone() then!